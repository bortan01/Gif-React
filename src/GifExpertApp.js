import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';


export const GifExpertApp = () => {
   const [categories, setCategories] = useState(['one punch', 'Samuray X', 'goku']);
   // const categories = ['one punch', 'Samuray X', 'goku'];
   // const handleAdd = () => {
   //    setCategories([...categories, 'one piece']);
   // setCategories(['one piece', ...categories]);
   // setCategories((cats) =>  [...cats, 'one piece'] );
   //};

   return (
      <>
         <h2>GifExpertApp</h2>
        <AddCategory/>
         <hr />
         <ol>
            {
               categories.map((cat) => {
                  return <li key={cat} >hola  {cat}</li>
               })
            }
         </ol>
      </>
   )
}
