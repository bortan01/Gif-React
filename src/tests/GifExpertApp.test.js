import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('pruebas en <GifExperApp/>', () => {

   test('probando el snapShop ', () => {
      const wraper = shallow(<GifExpertApp />);
      expect(wraper).toMatchSnapshot();
   });
   test('debe de mostrar una lista de categorias ', () => {
      const categories = ['one piece', 'to love ru'];
      const wraper = shallow(<GifExpertApp defaultCategories={categories} />);
      expect(wraper).toMatchSnapshot();
      expect(wraper.find('GifGrid').length).toBe(categories.length);

   })

});
