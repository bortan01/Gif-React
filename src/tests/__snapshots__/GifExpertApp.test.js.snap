// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExperApp/> debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one piece"
      key="one piece"
    />
    <GifGrid
      category="to love ru"
      key="to love ru"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExperApp/> probando el snapShop  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
